modules/taboolaBidAdapter.js

```diff
@@ -276,7 +276,7 @@ function getSiteProperties({publisherId}, refererInfo, ortb2) {
 function fillTaboolaReqData(bidderRequest, bidRequest, data) {
   const {refererInfo, gdprConsent = {}, uspConsent} = bidderRequest;
   const site = getSiteProperties(bidRequest.params, refererInfo, bidderRequest.ortb2);
-  deepSetValue(data, 'device.ua', navigator.userAgent);
+  deepSetValue(data, 'device', bidderRequest?.ortb2?.device || {ua: navigator.userAgent});
```

Comment:
I don't think this works; ua isn't in the current device object so this effectively deletes it, as the fallback will almost never happen

Reply:
Oh ua is there, so the fallback should just be deleted

https://github.com/prebid/Prebid.js/blob/66605617791f289988ca166428d348885a84f839/src/fpd/enrichment.js#L100

Reply:
Thanks Patrick, the fallback has been removed in the latest commit.